modules/core/src/arithm.simd.hpp

```diff
@@ -393,7 +393,7 @@ static void bin_loop(const T1* src1, size_t step1, const T1* src2, size_t step2,
 #if CV_SIMD
     typedef bin_loader<OP, T1, Tvec> ldr;
     enum {wide_step = Tvec::nlanes};
-    #if !CV_NEON && CV_SIMD_WIDTH == 16
+    #if !CV_NEON && !CV_MSA && CV_SIMD_WIDTH == 16
```

Comment:
In case data width is less than wide_step * 2, the performance would be better with this change. Otherwise, the performance shouldn't be better. Considering 'width' should be larger in most cases, this change would not be necessary. I will remove it.